doc/api/advanced_input.rst

```diff
@@ -0,0 +1,90 @@
+Advanced Input
+==============
+
+.. automodule:: simpleline.input.input_handler
+
+.. WARNING::
+    This section contains advanced input techniques which may lead to buggy code in your
+    program if they are not used correctly.
+
+The default technique to get user input is already described in :ref:`UIScreen <uiscreen_label>`,
+and that should be the preferred way to obtain user input. However, if the default technique is
+not enough for your situation then read the text below to find out how to implement a custom input.
+
+Input handler classes
+---------------------
+Input handler classes are classes created to obtain user input. If required
+:class:`InputHandler` can block an application until user input is received.
+
+This class can be instantiated everywhere in the code and used to ask for user input::
+
+    handler = InputHandler()
+    handler.get_input("Shut up and give me your input:")
+    handler.wait_for_input()
+    if handler.input_successful:
+        user_input = handler.value
+
+The :meth:`InputHandler.wait_on_input` method will block code processing until user input is
+received. The input should always be checked before processing.
+
+In case some other work needs to be done before user input is received, then pass
+a callback to the constructor of the :class:`InputHandler` class and do not use
+the :meth:`InputHandler.wait_on_input` method. However, the callback is using
+the :ref:`event loop<event_loops_label>` so the callback won't be called until event loop
+processing is active. If :ref:`concurrent input<concurrent_input_label>` is used then this
+callback might never get called!
+
+If what a user types must not be displayed, then the :class:`PasswordInputHandler` class
+should be used. It shares most of its implementation with the :class:`InputHandler` but overrides
+how to obtain the code and other small nuances. For more info look at the
```

Comment:
you can probably get rid of "and other small nuances". Are they worth mentioning? Then mention them. Otherwise what's the point?